spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -385,12 +400,21 @@ def main(argv=None):
                                                             group_funcs=group_funcs)
     metrics_agg_merged = merge_dict(metrics_agg)
     save_as_csv(metrics_agg_merged, file_out, fname_in=fname_segmentation, append=append)
-
-    # QC report (only show CSA for clarity)
+    # QC report (only for PMJ-based CSA)
     if path_qc is not None:
-        generate_qc(fname_segmentation, args=arguments, path_qc=os.path.abspath(path_qc), dataset=qc_dataset,
-                    subject=qc_subject, path_img=_make_figure(metrics_agg_merged, fit_results),
-                    process='sct_process_segmentation')
+        if fname_pmj is not None:
+            if arguments.qc_image is not None:
+                generate_qc(fname_in1=get_absolute_path(arguments.qc_image),
+                            fname_seg=[fname_ctl_smooth, fname_pmj, fname_mask_out, fname_ctl_smooth],  # Repeat centerline to flatten image along the centerline
+                            args=arguments,
```

Comment:
Passing `arguments` will cause all of the non-passed arguments to appear as well. (See screenshot.)

Looking at some other calls to `generate_sc`, it looks like `args=sys.argv[1:]` lets us pass only the args that were specified by the user.

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/c79a02021509da8b3c84878c6029a42a26a3d2d8/spinalcordtoolbox/scripts/sct_deepseg_gm.py#L127

Reply:
> Looking at some other calls to generate_sc, it looks like args=sys.argv[1:] lets us pass just the args that were specified by the user.

I tested it out in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3465/commits/a8170907e41d4d67bbe960f3ceb8d24865c2e2b8 and you are right, it lets us pass just the args that were specified by the user 🙂